numcodecs/gzip.py

```diff
@@ -44,11 +44,11 @@ def encode(self, buf):
             compressor.write(buf)
 
         try:
-            compressed = compressed.getbuffer()
+            compressed = ensure_ndarray(compressed.getbuffer())
```

Comment:
Do we want to ensure this is bytes here, for consistency with PY3?

Reply:
The idea here was to avoid a copy by just getting access to the buffer. That said, can understand how having this differ between Python 2/3 might be confusing. If we want to unify them, we should just use `compressed.getvalue()` in both cases.

Reply:
No worries. Thanks for clarifying. Should be fixed with the latest commit.